StockTrader/src/main/java/com/restResource/StockTrader/service/BuyTriggerService.java

```diff
@@ -0,0 +1,117 @@
+                for (;;) {
+
+                    try {
+                        Thread.sleep(5000);
+                    } catch(InterruptedException e) {
+                        throw new IllegalArgumentException(
+                                "Error with Thread.sleep in "+Thread.currentThread().getName());
+                    }
+
+                    Optional<BuyTrigger> buyStockSnapshot = buyTriggerRepository.findByUserIdAndStockSymbol(userId, stockSymbol);
+
+                    if (!buyStockSnapshot.isPresent()) { //we dont have an entry yet, so continue waiting
+                        continue;
+                    } else if (buyStockSnapshot.get().getStock_cost() != null) { //we already have a working buy trigger
+                        return;
+                    } else { //we have a trigger that is waiting for a cost target
+                        Integer amount = buyStockSnapshot.get().getStock_amount();
+                        if (accountRepository.removeFunds(userId, amount*stockCost, transactionNum, "TS1") == 0) {
+                            buyTriggerRepository.delete(buyStockSnapshot.get()); //remove it since it is not valid
+                            loggingService.logErrorEvent(
+                                    ErrorEventLog.builder()
+                                            .command(CommandType.SET_BUY_TRIGGER)
+                                            .username(userId)
+                                            .stockSymbol(stockSymbol)
+                                            .transactionNum(transactionNum)
+                                            .errorMessage("The amount parameter must be greater than zero")
+                                            .build());
+                            return;
+                        }
+                        buyStockSnapshot.get().setStock_cost(stockCost);
+                        buyTriggerRepository.save(buyStockSnapshot.get());
+                        break;
+                    }
+                }
+                //TODO technically can assign this from initialization
+                Integer cost;
+                Integer amount;
+                for (;;) {
+                    //TODO its possible avoid this db query, by saving the thread id to the db and deleteing the thread using that
+                    Optional<BuyTrigger> buyStockSnapshot = buyTriggerRepository.findByUserIdAndStockSymbol(userId, stockSymbol);
+                    if (buyStockSnapshot.isPresent()) {
+                        cost = buyStockSnapshot.get().getStock_cost();
+                        amount = buyStockSnapshot.get().getStock_amount();
+                    } else { //the trigger has been deleted -> not needed if we are removing the thread using the id
+                        return;
+                    }
+                    Quote quote = quoteService.getQuote(stockSymbol, userId, transactionNum);
+                    //TODO handle quote not existing
```

Comment:
Good catch here btw. I'm thinking quote service could return an optional and then we log an error here saying the stockSymbol doesn't exist, or we could do an initial check during the SET_BUY_AMOUNT command to verify that it exists, in which case we could probably just throw an error. What do you think? Also I should probably throw an error if can't connect to quote server, so we'll probably want try catches around quoteService calls